gpbft/gpbft.go

```diff
@@ -959,27 +959,27 @@ func (i *instance) rebroadcast() error {
 	// Note that the implementation here rebroadcasts more messages than FIP-0086
 	// strictly requires. Because, the cost of rebroadcasting additional messages is
 	// small compared to the reduction in need for rebroadcast.
-	var msgs []*MessageBuilder
+	var requests []*rebroadcastRequest
 	if i.broadcasted.decide != nil {
-		msgs = append(msgs, i.broadcasted.decide)
+		requests = append(requests, i.broadcasted.decide)
 	} else {
 		// We rebroadcast messages newest to oldest, except quality which we always
 		// rebroadcast first. That way, if we try to rebroadcast too many at once and some
 		// get dropped, chances are we've already rebroadcast the most useful messages.
-		msgs = append(msgs, i.broadcasted.previousRound...)
-		msgs = append(msgs, i.broadcasted.currentRound...)
+		requests = append(requests, i.broadcasted.previousRound...)
+		requests = append(requests, i.broadcasted.currentRound...)
 		if i.broadcasted.quality != nil {
-			msgs = append(msgs, i.broadcasted.quality)
+			requests = append(requests, i.broadcasted.quality)
 		}
-		slices.Reverse(msgs)
+		slices.Reverse(requests)
 	}
-	for _, mb := range msgs {
-		if err := i.participant.host.RequestBroadcast(mb); err != nil {
+	for _, request := range requests {
+		if err := i.participant.host.RequestRebroadcast(i.instanceID, request.round, request.phase); err != nil {
 			// Silently log the error and proceed. This is consistent with the behaviour of
 			// instance for regular broadcasts.
-			i.log("failed to request rebroadcast %s at round %d: %v", mb.Payload.Phase, mb.Payload.Round, err)
+			i.log("failed to request rebroadcast %s at round %d: %v", request.phase, request.round, err)
 		} else {
-			i.log("rebroadcasting %s at round %d for value %s", mb.Payload.Phase.String(), mb.Payload.Round, mb.Payload.Value)
+			i.log("rebroadcasting %s at round %d", request.phase, request.round)
 			metrics.reBroadcastCounter.Add(context.TODO(), 1)
 		}
 	}
```

Comment:
This works, but can't we also just not record anything? That is:

1. If we're in the decide phase, blindly request rebroadcast for decide.
2. If we're in any other phase, blindly request rebroadcast for each phase in the previous/current rounds, up to the current phase?

It doesn't matter if we actually sent any messages at those points in time.